tests/split_test.c

```diff
@@ -282,3 +282,50 @@ static int s_test_char_split_output_too_small_fn(struct aws_allocator *allocator
 
     return 0;
 }
+
+AWS_TEST_CASE(test_byte_cursor_next_split, s_test_byte_cursor_next_split)
+static int s_test_byte_cursor_next_split(struct aws_allocator *allocator, void *ctx) {
```

Comment:
😅 I thought we had explicit tests for this function

I guess we *implicitly* tested it since it's powering the other split functions.

This is a good addition